wwwverifier/src/main/java/com/google/sps/servlets/RequestServlet.java

```diff
@@ -0,0 +1,514 @@
+     */
+    private static Map<String, Map<String, Boolean>> createMdlItemsToRequest() {
+        Map<String, Map<String, Boolean>> mdlItemsToRequest = new HashMap<>();
+        Map<String, Boolean> mdlNsItems = new HashMap<>();
+        mdlNsItems.put("sex", false);
+        mdlNsItems.put("portrait", false);
+        mdlNsItems.put("given_name", false);
+        mdlNsItems.put("issue_date", false);
+        mdlNsItems.put("expiry_date", false);
+        mdlNsItems.put("family_name", false);
+        mdlNsItems.put("document_number", false);
+        mdlNsItems.put("issuing_authority", false);
+        mdlItemsToRequest.put(ServletConsts.MDL_NAMESPACE, mdlNsItems);
+        Map<String, Boolean> aamvaNsItems = new HashMap<>();
+        aamvaNsItems.put("DHS_compliance", false);
+        aamvaNsItems.put("EDL_credential", false);
+        mdlItemsToRequest.put(ServletConsts.AAMVA_NAMESPACE, aamvaNsItems);
+        return mdlItemsToRequest;
+    }
+
+    /**
+     * @return generated readerEngagement CBOR message, using EngagementGenerator
+     */
+    public static byte[] generateReaderEngagement(PublicKey publicKey, Key key) {
+        EngagementGenerator eg = new EngagementGenerator(publicKey,
+            EngagementGenerator.ENGAGEMENT_VERSION_1_1);
+        eg.addConnectionMethod(new ConnectionMethodHttp(ServletConsts.ABSOLUTE_URL + "/" 
+            + com.google.appengine.api.datastore.KeyFactory.keyToString(key)));
+        return eg.generate();
+    }
+
+    /**
+     * @return generated ephemeral reader key pair (containing a PublicKey and a PrivateKey)
+     */
+    private static KeyPair generateKeyPair() {
+        try {
+            KeyPairGenerator kpg = KeyPairGenerator.getInstance(ServletConsts.KEYGEN_INSTANCE);
+            ECGenParameterSpec ecSpec = new ECGenParameterSpec(ServletConsts.KEYGEN_CURVE);
+            kpg.initialize(ecSpec);
+            return kpg.generateKeyPair();
+        } catch (NoSuchAlgorithmException | InvalidAlgorithmParameterException e) {
+            throw new IllegalStateException("Error generating ephemeral key-pair", e);
+        }
+    }
+
+    /**
+     * @return byte array with data extracted from HTTP POST request @param request
+     */
+    private static byte[] getBytesFromRequest(HttpServletRequest request) {
+        byte[] arr = new byte[request.getContentLength()];
+        try {
+            request.getInputStream().read(arr);
+            return arr;
+        } catch (IOException e) {
+            throw new IllegalStateException("Error reading request body", e);
+        }
+    }
+
+    /**
+     * Parses DeviceResponse CBOR message and converts its contents into a JSON string.
+     * 
+     * @param messageData CBOR message containing DeviceResponse to be decoded and parsed
+     * @param key Unique key of the entity in Datastore that corresponds to the current
+     * session
+     * 
+     * @return byte array containing a termination message
+     */
```

Comment:
Comment needs update - it doesn't return a JSON string, it returns a SessionData with status 20 to indicate termination.